src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -457,25 +428,21 @@ bool emberAfOperationalCredentialsClusterOpCSRRequestCallback(EndpointId endpoin
     TLV::TLVWriter csrElementWriter;
     TLV::TLVType containerType;
 
-    // Fetch current fabric
-    FabricInfo * fabric = retrieveCurrentFabric();
-    VerifyOrExit(fabric != nullptr, status = EMBER_ZCL_STATUS_FAILURE);
+    VerifyOrExit(csr.Alloc(Crypto::kMAX_CSR_Length), err = CHIP_ERROR_NO_MEMORY);
 
-    VerifyOrExit(csr.Alloc(Crypto::kMAX_CSR_Length), status = EMBER_ZCL_STATUS_FAILURE);
-
-    if (fabric->GetOperationalKey() == nullptr)
+    if (gFabricBeingCommissioned.GetEphemeralKey() == nullptr)
     {
         Crypto::P256Keypair keypair;
         keypair.Initialize();
-        VerifyOrExit(fabric->SetOperationalKey(keypair) == CHIP_NO_ERROR, status = EMBER_ZCL_STATUS_FAILURE);
+        SuccessOrExit(err = gFabricBeingCommissioned.SetEphemeralKey(&keypair));
     }
 
-    err = fabric->GetOperationalKey()->NewCertificateSigningRequest(csr.Get(), csrLength);
+    err = gFabricBeingCommissioned.GetEphemeralKey()->NewCertificateSigningRequest(csr.Get(), csrLength);
     emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: NewCertificateSigningRequest returned %d", err);
-    VerifyOrExit(err == CHIP_NO_ERROR, status = EMBER_ZCL_STATUS_FAILURE);
-    VerifyOrExit(csrLength < UINT8_MAX, status = EMBER_ZCL_STATUS_FAILURE);
+    SuccessOrExit(err);
+    VerifyOrExit(csrLength < UINT8_MAX, err = CHIP_ERROR_INTERNAL);
```

Comment:
The CSR length can be > UINT8_MAX, in theory. Why is there such an error? Because of  Octet String not supporting > 255 bytes length?

Reply:
Yes, that's correct.

Reply:
When are we fixing this? It's been a thorn in our side for a long time. The NOC chain in AddOpCert can easily be > 255 bytes. What is happening with that?